packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,428 @@
+import Fieldset from 'terra-form/lib/Fieldset';
+import NumberField from 'terra-form/lib/NumberField';
+import DatePicker from 'terra-date-picker';
+import SelectField from 'terra-form/lib/SelectField';
+
+
+const propTypes = {
+  /**
+   * The ISO 8601 string representation of the birth date to calculate an onset date for the AGE precision.
+   * Also limits the earliest possible date that can be selected for an onset date for YEAR, MONTH, and DATE precision.
+   */
+  birthdate: PropTypes.string.isRequired,
+
+  /**
+   * The granularity of the onset date. AGE, YEAR, MONTH, and DATE are accepted.
+   */
+  granularity: PropTypes.oneOf(['AGE', 'YEAR', 'MONTH', 'DATE']),
+
+  /**
+   * Name of the granularity select. The name should be unique.
+   */
+  granularitySelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a granularity is selected.
+   * The first parameter is the event. The second parameter is the changed granularity value.
+   */
+  granularitySelectOnChange: PropTypes.func,
+
+  /**
+   * The precision of the onset date. ABOUT, BEFORE, AFTER, and UNKNOWN are accepted.
+   */
+  precision: PropTypes.oneOf(['ABOUT', 'BEFORE', 'AFTER', 'UNKNOWN']),
+
+  /**
+   * Name of the precision select. The name should be unique.
+   */
+  precisionSelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a precision is selected.
+   * The first parameter is the event. The second parameter is the changed precision value.
+   */
+  precisionSelectOnChange: PropTypes.func,
+
+  /**
+   * The ISO 8601 string representation of the onset date to view/modify. Defaults to current date.
+   */
+  onsetDate: PropTypes.string,
+
+  /**
+   * Name of the onsetDate input. The name should be unique.
+   */
+  onsetDateInputName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a onsetDate is changed.
+   * The first parameter is the event. The second parameter is the changed onsetDate value.
+   */
+  onsetDateInputOnChange: PropTypes.func,
+};
+
+const defaultProps = {
+  birthdate: undefined,
+  granularity: 'MONTH',
+  granularitySelectName: undefined,
+  granularitySelectOnChange: undefined,
+  precision: 'ABOUT',
+  precisionSelectName: undefined,
+  precisionSelectOnChange: undefined,
+  onsetDate: moment().format('YYYY-MM-DD'),
+  onsetDateInputName: undefined,
+  onsetDateInputOnChange: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class OnsetPicker extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate,
+      age: 1,
+      ageUnit: 'weeks',
+    };
```

Comment:
Should age be calculated initially? Also, do we want to default ageUnit to 'weeks'? Should that be an option in props?

Reply:
Reworked age calculations a bit. Now is calculated initially based on the onsetDate, and it updated correctly when switched to from other granularities. I don't think there needs to be any configuration options for age, it's just a granularity in the picker that we control.